typings/index.d.ts

```diff
@@ -1817,6 +1820,66 @@ export class MessageSelectMenu extends BaseMessageComponent {
   public toJSON(): APISelectMenuComponent;
 }
 
+export class Modal {
+  public constructor(data?: Modal | ModalOptions);
+  public components: MessageActionRow<ModalActionRowComponent>[];
+  public customId: string;
```

Comment:
```suggestion
  public customId: string | null;
```

this can be null before using `setCustomId`  same for other properties